drivers/counter/counter_mcux_pit.c

```diff
@@ -0,0 +1,209 @@
+
+	return 0;
+}
+
+static int mcux_pit_set_top_value(struct device *dev,
+				  const struct counter_top_cfg *cfg)
+{
+	const struct mcux_pit_config *config = dev->config_info;
+	struct mcux_pit_data *data = dev->driver_data;
+	pit_chnl_t channel = config->pit_channel;
+
+	if (cfg->ticks == 0) {
+		return -EINVAL;
+	}
+
+	data->top_callback = cfg->callback;
+	data->top_user_data = cfg->user_data;
+
+	if (config->base->CHANNEL[channel].TCTRL & PIT_TCTRL_TEN_MASK) {
+		/* Timer already enabled, check flags before resetting */
+		if (cfg->flags & COUNTER_TOP_CFG_DONT_RESET) {
+			return -ENOTSUP;
+		}
+		PIT_StopTimer(config->base, channel);
+		PIT_SetTimerPeriod(config->base, channel, cfg->ticks);
+		PIT_StartTimer(config->base, channel);
+	} else {
+		PIT_SetTimerPeriod(config->base, channel, cfg->ticks);
+	}
+
+	return 0;
+}
+
+static uint32_t mcux_pit_get_pending_int(struct device *dev)
+{
+	const struct mcux_pit_config *config = dev->config_info;
+	uint32_t mask = PIT_TFLG_TIF_MASK;
+	uint32_t flags;
+
+	flags = PIT_GetStatusFlags(config->base, config->pit_channel);
+
+	return ((flags & mask) == mask);
+}
+
+static uint32_t mcux_pit_get_top_value(struct device *dev)
+{
+	const struct mcux_pit_config *config = dev->config_info;
+	pit_chnl_t channel = config->pit_channel;
+
+	return config->base->CHANNEL[channel].LDVAL;
+}
+
+static uint32_t mcux_pit_get_max_relative_alarm(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/* Alarms not supported */
+	return 0;
+}
+
+static void mcux_pit_isr(void *arg)
+{
+	struct device *dev = arg;
+	const struct mcux_pit_config *config = dev->config_info;
+	struct mcux_pit_data *data = dev->driver_data;
+	uint32_t flags;
+
+	flags = PIT_GetStatusFlags(config->base, config->pit_channel);
+	PIT_ClearStatusFlags(config->base, config->pit_channel, flags);
+
+	if (data->top_callback) {
+		data->top_callback(dev, data->top_user_data);
+	}
+}
+
+static int mcux_pit_init(struct device *dev)
+{
+	struct mcux_pit_config *config =
+		(struct mcux_pit_config *)dev->config_info;
+	pit_config_t pit_config;
+
+	PIT_GetDefaultConfig(&pit_config);
+	pit_config.enableRunInDebug = config->enableRunInDebug;
+
+	PIT_Init(config->base, &pit_config);
+
+	config->irq_config_func(dev);
+
+	/*hard code to 1s*/
+	PIT_SetTimerPeriod(config->base, config->pit_channel,
+			   USEC_TO_COUNT(1000000U,
+					 CLOCK_GetFreq(kCLOCK_BusClk)));
+	config->info.freq = CLOCK_GetFreq(kCLOCK_BusClk);
+
+	return 0;
+}
+
+static const struct counter_driver_api mcux_pit_driver_api = {
+	.start = mcux_pit_start,
+	.stop = mcux_pit_stop,
+	.get_value = mcux_pit_get_value,
+	.set_top_value = mcux_pit_set_top_value,
+	.get_pending_int = mcux_pit_get_pending_int,
+	.get_top_value = mcux_pit_get_top_value,
+	.get_max_relative_alarm = mcux_pit_get_max_relative_alarm,
+};
+
+/*
+ * This driver is single-instance. If the devicetree contains multiple
+ * instances, this will fail and the driver needs to be revisited.
```

Comment:
Why not support multiple instances?

Reply:
in all MCU platform, we only have one PIT, I never test multiply PIT. maybe they are OK, but I copy it from lptmr and not sure I need remove it or not.